ansible/deploy.yml

```diff
@@ -44,10 +44,43 @@
              | default([]) | subelements('physical_networks') }}
 
 - hosts: libvirt
+  vars:
+    # Allocations are stored in the localhost's vars.
+    nodes: >-
+      {{ hostvars['localhost'].allocations.result[inventory_hostname]
+           | default([]) }}
   tasks:
-    - include_tasks: libvirt_create_vms.yml
+    - name: Create Libvirt VMs
+      include_tasks: libvirt_create_vms.yml
       vars:
-        # Allocations are stored in the localhost's vars.
-        vms: >-
-          {{ hostvars['localhost'].allocations.result[inventory_hostname]
-               | default([]) }}
+        vms: "{{ nodes }}"
+
+    - name: Check that enough ports are available for Virtual BMC
+      fail:
+        msg: >
+          {{ nodes | count }} nodes were specified to be added to Virtual BMC,
+          but only {{ ipmi_port_range_end - ipmi_port_range_start }} ports are
+          available for use by Virtual BMC.
+      when: >-
+          (nodes | count) > (ipmi_port_range_end - ipmi_port_range_start)
+
+    - name: Set up Virtual BMC daemon
+      include_role:
+        name: virtualbmc-daemon
+      vars:
+        vbmcd_virtualenv_path: "{{ virtualenv_path }}"
+        vbmcd_python_upper_contraints_url: "{{ python_upper_constraints_url }}"
+      when: (nodes | count) > 0
```

Comment:
Should this be moved with the host setup? We'd have to always run it, but that's not the end of the world.

Reply:
Seems reasonable.